src/service/resource.js

```diff
@@ -103,6 +104,15 @@ export class Resource {
       Resource.forElementOptional(element).updateOwner(owner);
     }
     element[OWNER_PROP_] = owner;
+    // Need to clear owner cache for all child elements
```

Comment:
Good point, it needs to be recursive. However I don't think `Resources#discoverResourcesForElement_` or `#discoverResourcesForArray_` will do the work here. In the case of `<owner> <child> <div> <grandchild>` `#discoverResourcesForElement_` will stop searching when it reaches `<child>`. Thus will never clear the cached value for `<grandchild>`. Also what if `<owner><child><div><grandchild><grandgrandchild>`?
A correct approach I can think of would be finding all the children, grandchildren .... (sorry I'm too lazy find the opposite word for ancestor 😄), but I assume it would be time consuming.